### PR TITLE
4.x - Update to phpunit 8

### PR DESCRIPTION
**Pull request by ADmad**
I had started working on this yesterday :slightly_smiling_face:.

Only warnings like the ones below remain. I am not sure about their replacement and they don't cause builds to fail so I left them for now.

```
attributeEqualTo() is deprecated and will be removed in PHPUnit 9.
attribute() is deprecated and will be removed in PHPUnit 9.
readAttribute() is deprecated and will be removed in PHPUnit 9.
getObjectAttribute() is deprecated and will be removed in PHPUnit 9.
```

### CODE REVIEW

**Comment by ADmad** [comment]
Psalm is unhappy with the use of various method/classes marked as internal. Just ignoring those for now.

https://scrutinizer-ci.com/g/cakephp/cakephp/inspections/3af03b26-556a-4307-8d57-820ecf3969f1

**Comment by ADmad** [comment]
The static analysis job failure is unrelated and fixed on `4.x`.

**Comment by dereuromark** [comment]
I fixed the phpstan inspection error in https://github.com/cakephp/cakephp/pull/13183/commits/155a22d9efda4d36516b42a78a3b454aff133bf1

File: tests/TestCase/Command/CacheCommandsTest.php [hunk 1]
```diff
@@ -42,7 +42,7 @@ public function setUp()
      *
      * @return void
      */
-    public function tearDown()
+    public function tearDown(): void
```
Comment (markstory) [inline tests/TestCase/Command/CacheCommandsTest.php]: Do we have sniffer rules that do this work or should we plan to add this to rector when we get to that point?

Comment (dereuromark) [reply tests/TestCase/Command/CacheCommandsTest.php]: Strict one of slevomatik could add those, but if enabled it also might add ones we dont want.
Better to run this manually from time to time maybe.

Comment (markstory) [reply tests/TestCase/Command/CacheCommandsTest.php]: I was more thinking about the upgrade path for applications and plugins, as I think all those builds/tests will now fail.

Comment (ADmad) [reply tests/TestCase/Command/CacheCommandsTest.php]: There are [rectors](https://github.com/rectorphp/rector/blob/master/docs/AllRectorsOverview.md#phpunit) available for updating the various assert methods.

**Pull request merged**